lib/linter/linter.js

```diff
@@ -866,93 +868,42 @@ function analyzeScope(ast, languageOptions, visitorKeys) {
-            loc: true,
-            range: true,
-            raw: true,
-            tokens: true,
-            comment: true,
-            eslintVisitorKeys: true,
-            eslintScopeManager: true,
-            filePath
-        }
-    );
-
-    /*
-     * Check for parsing errors first. If there's a parsing error, nothing
-     * else can happen. However, a parsing error does not throw an error
-     * from this method - it's just considered a fatal error message, a
-     * problem that ESLint identified just like any other.
-     */
-    try {
-        debug("Parsing:", filePath);
-        const parseResult = (typeof parser.parseForESLint === "function")
-            ? parser.parseForESLint(textToParse, parserOptions)
-            : { ast: parser.parse(textToParse, parserOptions) };
-
-        debug("Parsing successful:", filePath);
-        const ast = parseResult.ast;
-        const parserServices = parseResult.services || {};
-        const visitorKeys = parseResult.visitorKeys || evk.KEYS;
-
-        debug("Scope analysis:", filePath);
-        const scopeManager = parseResult.scopeManager || analyzeScope(ast, languageOptions, visitorKeys);
+function parse(file, language, languageOptions) {
 
-        debug("Scope analysis successful:", filePath);
+    const result = language.parse(file, { languageOptions });
 
+    if (result.ok) {
         return {
             success: true,
-
-            /*
-             * Save all values that `parseForESLint()` returned.
-             * If a `SourceCode` object is given as the first parameter instead of source code text,
-             * linter skips the parsing process and reuses the source code object.
-             * In that case, linter needs all the values that `parseForESLint()` returned.
-             */
-            sourceCode: new SourceCode({
-                text,
-                ast,
-                parserServices,
-                scopeManager,
-                visitorKeys
-            })
-        };
-    } catch (ex) {
-
-        // If the message includes a leading line number, strip it:
-        const message = `Parsing error: ${ex.message.replace(/^line \d+:/iu, "").trim()}`;
-
-        debug("%s\n%s", message, ex.stack);
-
-        return {
-            success: false,
-            error: {
-                ruleId: null,
-                fatal: true,
-                severity: 2,
-                message,
-                line: ex.lineNumber,
-                column: ex.column,
-                nodeType: null
-            }
+            sourceCode: language.createSourceCode(file, result, { languageOptions })
         };
     }
+
+    // if we made it to here there was an error
+    return {
+        success: false,
+        errors: result.errors.map(error => ({
+            ruleId: null,
+            nodeType: null,
+            fatal: true,
+            severity: 2,
+            message: error.message,
+            line: error.line,
+            column: error.column,
```

Comment:
Here we could also add offsets for `language.lineNumber` and `language.columnNumber`.